sdk/endpoint/switch.go

```diff
@@ -0,0 +1,61 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools/spanhelper"
+)
+
+// ConnectionEndpointMatcher is a function that accepts a connection and returns
+// the corresponding network service server.
+type ConnectionEndpointMatcher func(*connection.Connection) (networkservice.NetworkServiceServer, error)
```

Comment:
I do not think that the custom func should return network service.NetworkServiceServer.

```suggestion
type ConnectionEndpointMatcher func(*connection.Connection, config Configuration) bool
```